{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "01fbede0-4992-4703-993c-282e153cd394",
   "metadata": {},
   "source": [
    "# eDNA analysis test notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "de78c8c2-3b9c-46ff-9a2f-346fdeb2a60f",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd edna_analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "27297453-b016-4e8a-ab09-a86f658b3f40",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mcombined_reads\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0d2fbd50-5345-4e62-92c0-5dd14c6991df",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(fastp) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "conda activate fastp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d09c3a88-6cfc-4378-9b1e-cd4b9d78e8df",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(fastp) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir filtered_reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "85f1fb9e-6bfe-4e86-a94e-3690d987782b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Error to read gzip file\n",
      "ERROR: sequence and quality have different length:\n",
      "@5c25d512-ac3f-45bc-bb3d-bbe5868d873b runid=748ea7c9dba4efe93d87ff8436068ec76193362a read=19748 ch=398 start_time=2021-12-10T11:38:34.662417+00:00 flow_cell_id=FAV45916 protocol_group_id=invasive_edna_goldenpond sample_id=no_sample barcode=barcode96 barcode_alias=barcode96 parent_read_id=5c25d512-ac3f-45bc-bb3d-bbe5868d873b basecall_model_version_id=2021-09-08_dna_r10.4_minion_promethion_96_15fec0b3\n",
      "GTTATGCCTTTGGTTCGGTTCAATATTGTTAAGCATCCAACGGTCATGGTCCACCATGAGCACCTCCTGATTGCTGCATTATTCGCCAACATGGTGGCGAACATCAGCATGGTGGGCGATGCAGCTTGATCCTGATTTGTTGGCCTTGGTGTTAATTGTCTTTTGGTCGTGCATCAACCCCGACTGATTGGGATTGGCGTGGCGTTTGTTTTTCGGTGTGGCAACCGATGTGCATCAATCCGCTTTGCTGGGCCAGCACGCTTGTACACCGCTTTGGGGTTTATGGCCATCATGAGTGCATCGCCGCTTGTTGTCGTTCACCCGTGCCTTCGCAAGCCATGCAGGTCTTGCCCATGTTCGCTATTGCCCATC\n",
      "\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 838470\n",
      "total bases: 247823384\n",
      "Q20 bases: 99504567(40.1514%)\n",
      "Q30 bases: 35908640(14.4896%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 324382\n",
      "total bases: 128922585\n",
      "Q20 bases: 59271787(45.9747%)\n",
      "Q30 bases: 22106196(17.1469%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 324382\n",
      "reads failed due to low quality: 314094\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 199994\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0.0220589%\n",
      "\n",
      "JSON report: combined_reads/filtered_combined_edna.json\n",
      "HTML report: combined_reads/filtered_combined_edna.html\n",
      "\n",
      "fastp --in1 combined_reads/combined_edna.fastq.gz --out1 filtered_reads/filtered_combined_edna.fastq.gz --length_required 200 --qualified_quality_phred 12 --thread 16 --report_title Combined Q12+ reads >= 200bp --html combined_reads/filtered_combined_edna.html --json combined_reads/filtered_combined_edna.json \n",
      "fastp v0.20.0, time used: 36 seconds\n",
      "(fastp) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "fastp --in1 combined_reads/combined_edna.fastq.gz\\\n",
    "      --out1 filtered_reads/filtered_combined_edna.fastq.gz\\\n",
    "      --length_required 200\\\n",
    "      --qualified_quality_phred 12\\\n",
    "      --thread 16\\\n",
    "      --report_title \"Combined Q12+ reads >= 200bp\"\\\n",
    "      --html combined_reads/filtered_combined_edna.html\\\n",
    "      --json combined_reads/filtered_combined_edna.json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b2d0e664-8212-4ac4-be1c-72b6a0edb3e1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(porechop_abi) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "conda activate porechop_abi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d4fc3af7-7a40-44f5-84f4-b86b0f292c36",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mcombined_reads\u001b[0m  \u001b[01;34mfiltered_reads\u001b[0m\n",
      "(porechop_abi) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b70041dc-c77a-4ef1-9ad4-98cf42ffda58",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\u001b[1m\u001b[4mAb Initio Phase\u001b[0m\n",
      "Starting with a 10 run batch.\n",
      "Using config file:/home/exouser/.conda/envs/porechop_abi/lib/python3.10/site-packages/porechop_abi/ab_initio.config\n",
      "Command line:\n",
      " /home/exouser/.conda/envs/porechop_abi/lib/python3.10/site-packages/porechop_abi/approx_counter filtered_reads/filtered_combined_edna.fastq.gz -v 1 --config /home/exouser/.conda/envs/porechop_abi/lib/python3.10/site-packages/porechop_abi/ab_initio.config -o ./tmp/temp_approx_kmer_count -nt 8 -mr 10\n",
      "Kmer size:             16\n",
      "Sampled sequences:     40000\n",
      "Sampling length        100\n",
      "LC filter threshold:   1\n",
      "Adjusted LC threshold: 1\n",
      "Nb thread:             8\n",
      "Number of kept kmer:   500\n",
      "Number of runs:        10\n",
      "Verbosity level:       1\n",
      "\n",
      "A total of 10 runs will be performed.\n",
      "[10.5093 ms]\tParsing FASTA file\n",
      "[2528.03 ms]\tNumber of sequences found: 324382.\n",
      "Starting run number 1\n",
      "[2528.07 ms]\tWorking on sequence start.\n",
      "[9645.95 ms]\tWorking on sequence end.\n",
      "Starting run number 2\n",
      "[16639.3 ms]\tWorking on sequence start.\n",
      "[23755.5 ms]\tWorking on sequence end.\n",
      "Starting run number 3\n",
      "[30796 ms]\tWorking on sequence start.\n",
      "[37949.2 ms]\tWorking on sequence end.\n",
      "Starting run number 4\n",
      "[44918.6 ms]\tWorking on sequence start.\n",
      "[52064.2 ms]\tWorking on sequence end.\n",
      "Starting run number 5\n",
      "[59491.7 ms]\tWorking on sequence start.\n",
      "[66709.2 ms]\tWorking on sequence end.\n",
      "Starting run number 6\n",
      "[73692.3 ms]\tWorking on sequence start.\n",
      "[80904 ms]\tWorking on sequence end.\n",
      "Starting run number 7\n",
      "[87997.6 ms]\tWorking on sequence start.\n",
      "[95124.6 ms]\tWorking on sequence end.\n",
      "Starting run number 8\n",
      "[102117 ms]\tWorking on sequence start.\n",
      "[109284 ms]\tWorking on sequence end.\n",
      "Starting run number 9\n",
      "[116312 ms]\tWorking on sequence start.\n",
      "[123467 ms]\tWorking on sequence end.\n",
      "Starting run number 10\n",
      "[130491 ms]\tWorking on sequence start.\n",
      "[137649 ms]\tWorking on sequence end.\n",
      "Assembling run 1\n",
      "Assembling run 2\n",
      "Assembling run 3\n",
      "Assembling run 4\n",
      "Assembling run 5\n",
      "Assembling run 6\n",
      "Assembling run 7\n",
      "Assembling run 8\n",
      "Assembling run 9\n",
      "Assembling run 10\n",
      "/!\\\t The adapters currently found are not all identical.\n",
      "/!\\\t A consensus will be made.\n",
      "/!\\\t Launching 20 additional runs to build a consensus.\n",
      "Following with a 20 run batch.\n",
      "Using config file:/home/exouser/.conda/envs/porechop_abi/lib/python3.10/site-packages/porechop_abi/ab_initio.config\n",
      "Command line:\n",
      " /home/exouser/.conda/envs/porechop_abi/lib/python3.10/site-packages/porechop_abi/approx_counter filtered_reads/filtered_combined_edna.fastq.gz -v 1 --config /home/exouser/.conda/envs/porechop_abi/lib/python3.10/site-packages/porechop_abi/ab_initio.config -o ./tmp/temp_approx_kmer_count_sup -nt 8 -mr 20\n",
      "Kmer size:             16\n",
      "Sampled sequences:     40000\n",
      "Sampling length        100\n",
      "LC filter threshold:   1\n",
      "Adjusted LC threshold: 1\n",
      "Nb thread:             8\n",
      "Number of kept kmer:   500\n",
      "Number of runs:        20\n",
      "Verbosity level:       1\n",
      "\n",
      "A total of 20 runs will be performed.\n",
      "[3.45731 ms]\tParsing FASTA file\n",
      "[2532.63 ms]\tNumber of sequences found: 324382.\n",
      "Starting run number 1\n",
      "[2532.68 ms]\tWorking on sequence start.\n",
      "[9632.57 ms]\tWorking on sequence end.\n",
      "Starting run number 2\n",
      "[16646.5 ms]\tWorking on sequence start.\n",
      "[23762.3 ms]\tWorking on sequence end.\n",
      "Starting run number 3\n",
      "[30703.9 ms]\tWorking on sequence start.\n",
      "[37788.8 ms]\tWorking on sequence end.\n",
      "Starting run number 4\n",
      "[44706.7 ms]\tWorking on sequence start.\n",
      "[51821.4 ms]\tWorking on sequence end.\n",
      "Starting run number 5\n",
      "[58842.7 ms]\tWorking on sequence start.\n",
      "[65964.7 ms]\tWorking on sequence end.\n",
      "Starting run number 6\n",
      "[73025.2 ms]\tWorking on sequence start.\n",
      "[80160.1 ms]\tWorking on sequence end.\n",
      "Starting run number 7\n",
      "[87126 ms]\tWorking on sequence start.\n",
      "[94293 ms]\tWorking on sequence end.\n",
      "Starting run number 8\n",
      "[101362 ms]\tWorking on sequence start.\n",
      "[108489 ms]\tWorking on sequence end.\n",
      "Starting run number 9\n",
      "[115531 ms]\tWorking on sequence start.\n",
      "[122651 ms]\tWorking on sequence end.\n",
      "Starting run number 10\n",
      "[129628 ms]\tWorking on sequence start.\n",
      "[136691 ms]\tWorking on sequence end.\n",
      "Starting run number 11\n",
      "[143666 ms]\tWorking on sequence start.\n",
      "[150748 ms]\tWorking on sequence end.\n",
      "Starting run number 12\n",
      "[157693 ms]\tWorking on sequence start.\n",
      "[164795 ms]\tWorking on sequence end.\n",
      "Starting run number 13\n",
      "[171798 ms]\tWorking on sequence start.\n",
      "[178873 ms]\tWorking on sequence end.\n",
      "Starting run number 14\n",
      "[185826 ms]\tWorking on sequence start.\n",
      "[192903 ms]\tWorking on sequence end.\n",
      "Starting run number 15\n",
      "[199917 ms]\tWorking on sequence start.\n",
      "[206993 ms]\tWorking on sequence end.\n",
      "Starting run number 16\n",
      "[213957 ms]\tWorking on sequence start.\n",
      "[221028 ms]\tWorking on sequence end.\n",
      "Starting run number 17\n",
      "[227976 ms]\tWorking on sequence start.\n",
      "[235093 ms]\tWorking on sequence end.\n",
      "Starting run number 18\n",
      "[242123 ms]\tWorking on sequence start.\n",
      "[249225 ms]\tWorking on sequence end.\n",
      "Starting run number 19\n",
      "[256180 ms]\tWorking on sequence start.\n",
      "[263252 ms]\tWorking on sequence end.\n",
      "Starting run number 20\n",
      "[270209 ms]\tWorking on sequence start.\n",
      "[277348 ms]\tWorking on sequence end.\n",
      "Assembling run 1\n",
      "Assembling run 2\n",
      "Assembling run 3\n",
      "Assembling run 4\n",
      "Assembling run 5\n",
      "Assembling run 6\n",
      "Assembling run 7\n",
      "Assembling run 8\n",
      "Assembling run 9\n",
      "Assembling run 10\n",
      "Assembling run 11\n",
      "Assembling run 12\n",
      "Assembling run 13\n",
      "Assembling run 14\n",
      "Assembling run 15\n",
      "Assembling run 16\n",
      "Assembling run 17\n",
      "Assembling run 18\n",
      "Assembling run 19\n",
      "Assembling run 20\n",
      "Consensus step done\n",
      "\n",
      "Rebuild adapters:\n",
      "\n",
      "Start\n",
      "Consensus_1_start_(100.0%)\n",
      "GTGTCATGTAACCTACTTCGTTCAGTTACGTATTGCTAAGGTTAAGATTGTCCTCAAACTGCCACCTACCAGCACCTG\n",
      "\n",
      "End\n",
      "Consensus_1_end_(100.0%)\n",
      "AGGTGCTGGTAGGTGGCAGTTTGAGGACAATCTTAACCTTAGCAAT\n",
      "\n",
      "Building consensus adapter objects\n",
      "The inference of adapters sequence is done.\n",
      "\n",
      "\u001b[1m\u001b[4mLoading reads\u001b[0m\n",
      "filtered_reads/filtered_combined_edna.fastq.gz\n",
      "324,382 reads loaded\n",
      "\n",
      "\n",
      "\u001b[1m\u001b[4mLooking for known adapter sets\u001b[0m\n",
      "10,000 / 10,000 (100.0%)\n",
      "                                        Best               \u001b[0m\n",
      "                                        read       Best    \u001b[0m\n",
      "                                        start      read end\u001b[0m\n",
      "  \u001b[4mSet                                   %ID        %ID     \u001b[0m\n",
      "  \u001b[32mSQK-NSK007                               100.0       81.8\u001b[0m\n",
      "  Rapid                                     65.5        0.0\n",
      "  RBK004_upstream                           80.0        0.0\n",
      "  SQK-MAP006                                82.1       78.3\n",
      "  SQK-MAP006 short                          74.1       72.0\n",
      "  PCR adapters 1                            79.2       78.3\n",
      "  PCR adapters 2                            81.8       78.3\n",
      "  PCR adapters 3                            80.0       78.3\n",
      "  1D^2 part 1                               72.4       74.1\n",
      "  1D^2 part 2                               88.2       74.2\n",
      "  cDNA SSP                                  72.5       71.1\n",
      "  Barcode 1 (reverse)                       76.9       83.3\n",
      "  Barcode 2 (reverse)                       77.8       76.9\n",
      "  Barcode 3 (reverse)                       73.1       76.9\n",
      "  Barcode 4 (reverse)                       75.0       75.0\n",
      "  Barcode 5 (reverse)                       75.0       76.0\n",
      "  Barcode 6 (reverse)                       75.0       76.0\n",
      "  Barcode 7 (reverse)                       76.0       76.9\n",
      "  Barcode 8 (reverse)                       76.0       75.0\n",
      "  Barcode 9 (reverse)                       74.1       76.0\n",
      "  Barcode 10 (reverse)                      77.8       76.0\n",
      "  Barcode 11 (reverse)                      75.0       76.0\n",
      "  Barcode 12 (reverse)                      76.9       76.0\n",
      "  Barcode 1 (forward)                       83.3       77.8\n",
      "  Barcode 2 (forward)                       77.8       77.8\n",
      "  Barcode 3 (forward)                       76.9       79.2\n",
      "  Barcode 4 (forward)                       76.0       76.0\n",
      "  Barcode 5 (forward)                       80.0       74.1\n",
      "  Barcode 6 (forward)                       76.0       76.0\n",
      "  Barcode 7 (forward)                       75.0       76.0\n",
      "  Barcode 8 (forward)                       73.1       75.0\n",
      "  Barcode 9 (forward)                       75.0       75.0\n",
      "  Barcode 10 (forward)                      76.0       79.2\n",
      "  Barcode 11 (forward)                      76.0       75.0\n",
      "  Barcode 12 (forward)                      76.0       76.9\n",
      "  Barcode 13 (forward)                      76.0       79.2\n",
      "  Barcode 14 (forward)                      80.0       76.0\n",
      "  Barcode 15 (forward)                      76.0       76.0\n",
      "  Barcode 16 (forward)                      79.2       79.2\n",
      "  Barcode 17 (forward)                      74.1       75.9\n",
      "  Barcode 18 (forward)                      75.0       78.6\n",
      "  Barcode 19 (forward)                      76.0       76.9\n",
      "  Barcode 20 (forward)                      76.0       76.0\n",
      "  Barcode 21 (forward)                      74.1       74.1\n",
      "  Barcode 22 (forward)                      79.3       79.2\n",
      "  Barcode 23 (forward)                      72.0       74.1\n",
      "  Barcode 24 (forward)                      80.8       83.3\n",
      "  Barcode 25 (forward)                      76.0       79.2\n",
      "  Barcode 26 (forward)                      76.0       79.2\n",
      "  Barcode 27 (forward)                      76.0       76.9\n",
      "  Barcode 28 (forward)                      75.9       76.7\n",
      "  Barcode 29 (forward)                      76.9       76.9\n",
      "  Barcode 30 (forward)                      75.0       76.9\n",
      "  Barcode 31 (forward)                      76.9       79.2\n",
      "  Barcode 32 (forward)                      73.1       76.9\n",
      "  Barcode 33 (forward)                      76.0       76.9\n",
      "  Barcode 34 (forward)                      76.0       75.0\n",
      "  Barcode 35 (forward)                      80.8       79.2\n",
      "  Barcode 36 (forward)                      76.0       75.0\n",
      "  Barcode 37 (forward)                      75.0       76.0\n",
      "  Barcode 38 (forward)                      75.0       76.0\n",
      "  Barcode 39 (forward)                      75.0       76.0\n",
      "  Barcode 40 (forward)                      75.0       81.5\n",
      "  Barcode 41 (forward)                      83.3       80.0\n",
      "  Barcode 42 (forward)                      76.0       76.0\n",
      "  Barcode 43 (forward)                      76.0       76.0\n",
      "  Barcode 44 (forward)                      73.3       75.0\n",
      "  Barcode 45 (forward)                      77.8       79.2\n",
      "  Barcode 46 (forward)                      76.0       80.0\n",
      "  Barcode 47 (forward)                      73.1       83.3\n",
      "  Barcode 48 (forward)                      76.0       76.0\n",
      "  Barcode 49 (forward)                      79.2       80.0\n",
      "  Barcode 50 (forward)                      76.9       76.0\n",
      "  Barcode 51 (forward)                      76.0       76.0\n",
      "  Barcode 52 (forward)                      77.8       80.8\n",
      "  Barcode 53 (forward)                      75.0       76.9\n",
      "  Barcode 54 (forward)                      75.0       76.0\n",
      "  Barcode 55 (forward)                      75.0       76.9\n",
      "  Barcode 56 (forward)                      76.0       79.2\n",
      "  Barcode 57 (forward)                      76.9       80.0\n",
      "  Barcode 58 (forward)                      75.0       76.0\n",
      "  Barcode 59 (forward)                      75.0       76.0\n",
      "  Barcode 60 (forward)                      75.0       75.0\n",
      "  Barcode 61 (forward)                      75.0       77.8\n",
      "  Barcode 62 (forward)                      77.8       76.9\n",
      "  Barcode 63 (forward)                      75.0       74.1\n",
      "  Barcode 64 (forward)                      76.0       76.9\n",
      "  Barcode 65 (forward)                      76.0       79.2\n",
      "  Barcode 66 (forward)                      76.0       75.0\n",
      "  Barcode 67 (forward)                      76.9       76.0\n",
      "  Barcode 68 (forward)                      79.2       79.2\n",
      "  Barcode 69 (forward)                      74.1       75.0\n",
      "  Barcode 70 (forward)                      75.0       75.9\n",
      "  Barcode 71 (forward)                      76.0       76.0\n",
      "  Barcode 72 (forward)                      76.0       77.8\n",
      "  Barcode 73 (forward)                      80.8       80.8\n",
      "  Barcode 74 (forward)                      84.0       79.3\n",
      "  Barcode 75 (forward)                      75.0       75.0\n",
      "  Barcode 76 (forward)                      76.9       80.0\n",
      "  Barcode 77 (forward)                      74.1       80.8\n",
      "  Barcode 78 (forward)                      80.0       76.9\n",
      "  Barcode 79 (forward)                      76.0       76.9\n",
      "  Barcode 80 (forward)                      76.0       76.0\n",
      "  Barcode 81 (forward)                      72.0       76.9\n",
      "  Barcode 82 (forward)                      73.1       76.0\n",
      "  Barcode 83 (forward)                      75.0       73.1\n",
      "  Barcode 84 (forward)                      77.8       79.2\n",
      "  Barcode 85 (forward)                      76.9       75.0\n",
      "  Barcode 86 (forward)                      73.1       76.0\n",
      "  Barcode 87 (forward)                      75.0       76.0\n",
      "  Barcode 88 (forward)                      76.0       77.8\n",
      "  Barcode 89 (forward)                      79.2       79.2\n",
      "  Barcode 90 (forward)                      76.9       76.0\n",
      "  Barcode 91 (forward)                      76.0       76.0\n",
      "  Barcode 92 (forward)                      76.9       76.0\n",
      "  Barcode 93 (forward)                      75.0       80.0\n",
      "  \u001b[32mBarcode 94 (forward)                     100.0      100.0\u001b[0m\n",
      "  \u001b[32mBarcode 95 (forward)                      92.0       96.0\u001b[0m\n",
      "  \u001b[32mBarcode 96 (forward)                      83.3      100.0\u001b[0m\n",
      "  \u001b[32mConsensus_1_start_(100.0%)_adapter        97.5        0.0\u001b[0m\n",
      "  \u001b[32mConsensus_1_end_(100.0%)_adapter           0.0       97.9\u001b[0m\n",
      "\n",
      "\n",
      "\u001b[1m\u001b[4mTrimming adapters from read ends\u001b[0m\n",
      "            SQK-NSK007_Y_Top: \u001b[31mAATGTACTTCGTTCAGTTACGTATTGCT\u001b[0m\n",
      "         SQK-NSK007_Y_Bottom: \u001b[31mGCAATACGTAACTGAACGAAGT\u001b[0m\n",
      "                        BC94: \u001b[31mGATTGTCCTCAAACTGCCACCTAC\u001b[0m\n",
      "                    BC94_rev: \u001b[31mGTAGGTGGCAGTTTGAGGACAATC\u001b[0m\n",
      "                        BC95: \u001b[31mCCTGTCTGGAAGAAGAATGGACTT\u001b[0m\n",
      "                    BC95_rev: \u001b[31mAAGTCCATTCTTCTTCCAGACAGG\u001b[0m\n",
      "                        BC96: \u001b[31mCTGAACGGTCATAGAGTCCACCAT\u001b[0m\n",
      "                    BC96_rev: \u001b[31mATGGTGGACTCTATGACCGTTCAG\u001b[0m\n",
      "  Consensus_1_start_(100.0%): \u001b[31mGTGTCATGTAACCTACTTCGTTCAGTTACGTATTGCTAAGGTTAAGATTGTCCTCAAACTGCCACCTACCAGCACCTG\n",
      "\u001b[0m\n",
      "    Consensus_1_end_(100.0%): \u001b[31mAGGTGCTGGTAGGTGGCAGTTTGAGGACAATCTTAACCTTAGCAAT\n",
      "\u001b[0m\n",
      "\n",
      "324,382 / 324,382 (100.0%)\n",
      "\n",
      "319,283 / 324,382 reads had adapters trimmed from their start (23,426,018 bp removed)\n",
      "155,892 / 324,382 reads had adapters trimmed from their end (6,195,761 bp removed)\n",
      "\n",
      "\n",
      "\u001b[1m\u001b[4mSplitting reads containing middle adapters\u001b[0m\n",
      "324,382 / 324,382 (100.0%)\n",
      "\n",
      "1,469 / 324,382 reads were split based on middle adapters\n",
      "\n",
      "\n",
      "\u001b[1m\u001b[4mSaving trimmed reads to file\u001b[0m\n",
      "pigz found - using it to compress instead of gzip\n",
      "\n",
      "Saved result to /home/exouser/edna_analysis/filtered_reads/chop_filtered_combined_edna.fastq.gz\n",
      "\n",
      "(porechop_abi) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "porechop_abi --ab_initio -i filtered_reads/filtered_combined_edna.fastq.gz -o filtered_reads/chop_filtered_combined_edna.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "af59a558-aa40-4728-aba4-c7c680050a56",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(amplicon_sorter) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "conda activate amplicon_sorter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "48e8fe3b-9a1d-4c93-bd89-5488f6c47c64",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(amplicon_sorter) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cd ~/amplicon_sorter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "af2e031e-fc28-40e9-baf1-d7a207964577",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Amplicon_sorter_manual.pdf  amplicon_sorter.py  \u001b[0m\u001b[01;35mreadlengthhistogram.jpg\u001b[0m\n",
      "README.md                   \u001b[01;35mampliconsorter.jpg\u001b[0m\n",
      "(amplicon_sorter) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "7c5454e8-591b-4726-88a0-6b87f6bd762b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: amplicon_sorter.py [-h] -i INPUT [-min MINLENGTH] [-max MAXLENGTH]\n",
      "                          [-maxr MAXREADS] [-ar] [-np NPROCESSES]\n",
      "                          [-sg SIMILAR_GENES] [-ssg SIMILAR_SPECIES_GROUPS]\n",
      "                          [-ss SIMILAR_SPECIES] [-sc SIMILAR_CONSENSUS]\n",
      "                          [-ldc LENGTH_DIFF_CONSENSUS] [-sfq] [-ra] [-a]\n",
      "                          [-o OUTPUTFOLDER] [-ho] [-so]\n",
      "amplicon_sorter.py: error: the following arguments are required: -i/--input\n",
      "(amplicon_sorter) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "python3 amplicon_sorter.py "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "af81959e-c8f5-46b3-b8d4-c9322a1f453e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(amplicon_sorter) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir ~/edna_analysis/sorted_amplicons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "05186f3c-16bd-4145-a3d6-9b85482ef502",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(amplicon_sorter) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cd ~/edna_analysis/filtered_reads && gzip -d chop_filtered_combined_edna.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "4acd6f9f-e8a5-4bf3-b9d4-e02f9c5794f9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(amplicon_sorter) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cd ~/amplicon_sorter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "456ee320-861a-4680-a163-1d92322d1264",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading chop_filtered_combined_edna.fastq\n",
      "chop_filtered_combined_edna.fastq contains 322929 reads.\n",
      "--> Reading random 10000 out of 103344 sequences longer than 300bp\n",
      "processing: file_0.todo\n",
      "Estimating the ssg value for this dataset\n",
      "-> Estimated ssg = 90\n",
      "Filtering compared sequences for best hits and create groups\n",
      "Creating gene groups\n",
      "--> Number of groups before merge: 975\n",
      "--> Number of groups after merge: 508\n",
      "-> Merging based on consensus of 50 reads per group\n",
      "...making consensuses file_0.todo\n",
      "...making consensuses file_1.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 508\n",
      "--> Number of groups after merge: 79\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 79\n",
      "--> Number of groups after merge: 79\n",
      "--> Number of groups after removing groups with less than 5 sequences: 27\n",
      "  chop_filtered_combined_edna_0.group contains 3778 sequences (39.42%)\n",
      "  chop_filtered_combined_edna_1.group contains 159 sequences (1.66%)\n",
      "  chop_filtered_combined_edna_2.group contains 62 sequences (0.65%)\n",
      "  chop_filtered_combined_edna_3.group contains 458 sequences (4.78%)\n",
      "  chop_filtered_combined_edna_4.group contains 38 sequences (0.4%)\n",
      "  chop_filtered_combined_edna_5.group contains 77 sequences (0.8%)\n",
      "  chop_filtered_combined_edna_6.group contains 53 sequences (0.55%)\n",
      "  chop_filtered_combined_edna_7.group contains 7 sequences (0.07%)\n",
      "  chop_filtered_combined_edna_8.group contains 35 sequences (0.37%)\n",
      "  chop_filtered_combined_edna_9.group contains 25 sequences (0.26%)\n",
      "  chop_filtered_combined_edna_10.group contains 6 sequences (0.06%)\n",
      "  chop_filtered_combined_edna_11.group contains 11 sequences (0.11%)\n",
      "  chop_filtered_combined_edna_12.group contains 28 sequences (0.29%)\n",
      "  chop_filtered_combined_edna_13.group contains 17 sequences (0.18%)\n",
      "  chop_filtered_combined_edna_14.group contains 10 sequences (0.1%)\n",
      "  chop_filtered_combined_edna_15.group contains 19 sequences (0.2%)\n",
      "  chop_filtered_combined_edna_16.group contains 6 sequences (0.06%)\n",
      "  chop_filtered_combined_edna_17.group contains 7 sequences (0.07%)\n",
      "  chop_filtered_combined_edna_18.group contains 45 sequences (0.47%)\n",
      "  chop_filtered_combined_edna_19.group contains 7 sequences (0.07%)\n",
      "  chop_filtered_combined_edna_20.group contains 14 sequences (0.15%)\n",
      "  chop_filtered_combined_edna_21.group contains 11 sequences (0.11%)\n",
      "  chop_filtered_combined_edna_22.group contains 6 sequences (0.06%)\n",
      "  chop_filtered_combined_edna_23.group contains 8 sequences (0.08%)\n",
      "  chop_filtered_combined_edna_24.group contains 9 sequences (0.09%)\n",
      "  chop_filtered_combined_edna_25.group contains 6 sequences (0.06%)\n",
      "  chop_filtered_combined_edna_26.group contains 8 sequences (0.08%)\n",
      "4910/9584 sequences assigned in groups (51.23%)\n",
      "reading chop_filtered_combined_edna_1.group containing 159 sequences.\n",
      "--> Number of groups before merge: 19\n",
      "--> Number of groups after merge: 18\n",
      "--> Number of groups after removing groups with less than 4 sequences: 2\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group--> Comparing the rest of the sequences with the consensus sequences\n",
      "chop_filtered_combined_edna_1.group----> similarity = 0.95\n",
      "chop_filtered_combined_edna_1.group----> 212 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 1 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 190 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_1.group----> similarity = 0.94\n",
      "chop_filtered_combined_edna_1.group----> 2 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 200 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 2 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 2\n",
      "--> Number of groups after merge: 1\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 115 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 1 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "---finetune group 0 cycle 1\n",
      "---finetune group 0 cycle 2\n",
      "chop_filtered_combined_edna_1.group----> 123 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> similarity = 0.93\n",
      "chop_filtered_combined_edna_1.group----> 8 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 115 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 3 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 112 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_1.group----> similarity = 0.92\n",
      "chop_filtered_combined_edna_1.group----> 10 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 93 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 1 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 92 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_1.group----> similarity = 0.91\n",
      "chop_filtered_combined_edna_1.group----> 14 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 74 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 2 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 72 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_1.group----> similarity = 0.9\n",
      "chop_filtered_combined_edna_1.group----> 12 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 44 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 9 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 35 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_1.group----> similarity = 0.89\n",
      "chop_filtered_combined_edna_1.group----> 3 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 32 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_1.group----> similarity = 0.88\n",
      "chop_filtered_combined_edna_1.group----> 8 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 24 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> no sequences passed criteria\n",
      "---finetune group 0 cycle 1\n",
      "---finetune group 0 cycle 2\n",
      "---finetune group 0 cycle 3\n",
      "chop_filtered_combined_edna_1.group----> 32 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> similarity = 0.87\n",
      "chop_filtered_combined_edna_1.group----> 27 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 13 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 7 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 6 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 1 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> similarity = 0.86\n",
      "chop_filtered_combined_edna_1.group----> 1 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> 5 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_1.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_1.group----> similarity = 0.85\n",
      "chop_filtered_combined_edna_1.group----> 1 sequences added to the groups\n",
      "chop_filtered_combined_edna_1.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_13.group containing 17 sequences.\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_15.group containing 19 sequences.\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_11.group containing 11 sequences.\n",
      "--> Number of groups before merge: 2\n",
      "--> Number of groups after merge: 2\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_6.group containing 53 sequences.\n",
      "--> Number of groups before merge: 6\n",
      "--> Number of groups after merge: 6\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_2.group containing 62 sequences.\n",
      "--> Number of groups before merge: 7\n",
      "--> Number of groups after merge: 7\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_10.group containing 6 sequences.\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_12.group containing 28 sequences.\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_7.group containing 7 sequences.\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_21.group containing 11 sequences.\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_17.group containing 7 sequences.\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_25.group containing 6 sequences.\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_16.group containing 6 sequences.\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_22.group containing 6 sequences.\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_9.group containing 25 sequences.\n",
      "--> Number of groups before merge: 3\n",
      "--> Number of groups after merge: 3\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_26.group containing 8 sequences.\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_5.group containing 77 sequences.\n",
      "--> Number of groups before merge: 9\n",
      "--> Number of groups after merge: 9\n",
      "--> Number of groups after removing groups with less than 4 sequences: 1\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group--> Comparing the rest of the sequences with the consensus sequences\n",
      "chop_filtered_combined_edna_5.group----> similarity = 0.95\n",
      "chop_filtered_combined_edna_5.group----> 68 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 2 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 66 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_5.group----> similarity = 0.94\n",
      "chop_filtered_combined_edna_5.group----> 6 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 63 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 2 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 61 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> no sequences passed criteria\n",
      "---finetune group 0 cycle 1\n",
      "---finetune group 0 cycle 2\n",
      "chop_filtered_combined_edna_5.group----> 66 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> similarity = 0.93\n",
      "chop_filtered_combined_edna_5.group----> 12 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 55 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_5.group----> similarity = 0.92\n",
      "chop_filtered_combined_edna_5.group----> 8 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 47 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 3 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 44 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_5.group----> similarity = 0.91\n",
      "chop_filtered_combined_edna_5.group----> 8 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 36 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_5.group----> similarity = 0.9\n",
      "chop_filtered_combined_edna_5.group----> 3 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 33 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_5.group----> similarity = 0.89\n",
      "chop_filtered_combined_edna_5.group----> 5 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 28 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_5.group----> similarity = 0.88\n",
      "chop_filtered_combined_edna_5.group----> 2 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 26 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> no sequences passed criteria\n",
      "---finetune group 0 cycle 1\n",
      "---finetune group 0 cycle 2\n",
      "---finetune group 0 cycle 3\n",
      "chop_filtered_combined_edna_5.group----> 66 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> similarity = 0.87\n",
      "chop_filtered_combined_edna_5.group----> 47 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 19 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 1 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 18 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 4 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> similarity = 0.86\n",
      "chop_filtered_combined_edna_5.group----> 10 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 8 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 3 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 5 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> 1 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_5.group----> similarity = 0.85\n",
      "chop_filtered_combined_edna_5.group----> 2 sequences added to the groups\n",
      "chop_filtered_combined_edna_5.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_0.group containing 3778 sequences.\n",
      "--> Number of groups before merge: 262\n",
      "--> Number of groups after merge: 150\n",
      "--> Number of groups after removing groups with less than 4 sequences: 49\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group--> Comparing the rest of the sequences with the consensus sequences\n",
      "chop_filtered_combined_edna_0.group----> similarity = 0.95\n",
      "chop_filtered_combined_edna_0.group----> 25603 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 11 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 25522 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 5 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 49\n",
      "--> Number of groups after merge: 13\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 13\n",
      "--> Number of groups after merge: 12\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 12\n",
      "--> Number of groups after merge: 12\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 6037 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 1 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> similarity = 0.94\n",
      "chop_filtered_combined_edna_0.group----> 13 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 5975 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 3 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 12\n",
      "--> Number of groups after merge: 12\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 5979 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> no sequences passed criteria\n",
      "---finetune group 0 cycle 1\n",
      "---finetune group 0 cycle 2\n",
      "---finetune group 0 cycle 3\n",
      "---finetune group 1 cycle 1\n",
      "---finetune group 1 cycle 2\n",
      "---finetune group 1 cycle 3\n",
      "---finetune group 1 cycle 4\n",
      "---finetune group 1 cycle 5\n",
      "---finetune group 1 cycle 6\n",
      "---finetune group 1 cycle 7\n",
      "---finetune group 1 cycle 8\n",
      "---finetune group 1 cycle 9\n",
      "---finetune group 1 cycle 10\n",
      "---finetune group 2 cycle 1\n",
      "---finetune group 2 cycle 2\n",
      "---finetune group 3 cycle 1\n",
      "---finetune group 3 cycle 2\n",
      "---finetune group 4 cycle 1\n",
      "---finetune group 4 cycle 2\n",
      "---finetune group 5 cycle 1\n",
      "---finetune group 5 cycle 2\n",
      "---finetune group 6 cycle 1\n",
      "---finetune group 6 cycle 2\n",
      "---finetune group 6 cycle 3\n",
      "---finetune group 7 cycle 1\n",
      "---finetune group 7 cycle 2\n",
      "---finetune group 7 cycle 3\n",
      "---finetune group 8 cycle 1\n",
      "---finetune group 8 cycle 2\n",
      "---finetune group 9 cycle 1\n",
      "---finetune group 9 cycle 2\n",
      "---finetune group 10 cycle 1\n",
      "---finetune group 10 cycle 2\n",
      "---finetune group 11 cycle 1\n",
      "---finetune group 11 cycle 2\n",
      "chop_filtered_combined_edna_0.group----> 6700 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> similarity = 0.93\n",
      "chop_filtered_combined_edna_0.group----> 420 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 5624 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 62 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 7\n",
      "--> Number of groups after merge: 5\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 5\n",
      "--> Number of groups after merge: 5\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 3685 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 9 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> similarity = 0.92\n",
      "chop_filtered_combined_edna_0.group----> 145 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 3402 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 8 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 5\n",
      "--> Number of groups after merge: 5\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 3398 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 1 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> similarity = 0.91\n",
      "chop_filtered_combined_edna_0.group----> 204 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 2962 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 62 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 5\n",
      "--> Number of groups after merge: 5\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 2823 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 23 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> similarity = 0.9\n",
      "chop_filtered_combined_edna_0.group----> 236 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 2336 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 46 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 5\n",
      "--> Number of groups after merge: 5\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 2276 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 3 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> similarity = 0.89\n",
      "chop_filtered_combined_edna_0.group----> 129 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 2003 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 23 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 5\n",
      "--> Number of groups after merge: 5\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 1961 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 15 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> similarity = 0.88\n",
      "chop_filtered_combined_edna_0.group----> 116 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 1759 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 14 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 5\n",
      "--> Number of groups after merge: 5\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 1751 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 23 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "---finetune group 0 cycle 1\n",
      "---finetune group 0 cycle 2\n",
      "---finetune group 0 cycle 3\n",
      "---finetune group 1 cycle 1\n",
      "---finetune group 1 cycle 2\n",
      "---finetune group 2 cycle 1\n",
      "---finetune group 2 cycle 2\n",
      "---finetune group 2 cycle 3\n",
      "---finetune group 2 cycle 4\n",
      "---finetune group 2 cycle 5\n",
      "---finetune group 3 cycle 1\n",
      "---finetune group 3 cycle 2\n",
      "---finetune group 3 cycle 3\n",
      "---finetune group 3 cycle 4\n",
      "---finetune group 3 cycle 5\n",
      "---finetune group 4 cycle 1\n",
      "---finetune group 4 cycle 2\n",
      "---finetune group 4 cycle 3\n",
      "chop_filtered_combined_edna_0.group----> 4371 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> similarity = 0.87\n",
      "chop_filtered_combined_edna_0.group----> 1522 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 1690 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 24 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 5\n",
      "--> Number of groups after merge: 4\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 4\n",
      "--> Number of groups after merge: 4\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 1063 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 5 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> similarity = 0.86\n",
      "chop_filtered_combined_edna_0.group----> 42 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 1008 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 7 sequences added to the groups\n",
      "chop_filtered_combined_edna_0.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "--> Number of groups before merge: 4\n",
      "--> Number of groups after merge: 4\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_0.group----> 997 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "--> Number of groups after merge: 1\n",
      "...making consensuses file_0.todo\n",
      "...comparing consensuses file_0.todo\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 248 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_3.group----> similarity = 0.92\n",
      "chop_filtered_combined_edna_3.group----> 29 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 219 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 1 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 213 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 2 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> similarity = 0.91\n",
      "chop_filtered_combined_edna_3.group----> 30 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 188 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 3 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 183 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 2 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> similarity = 0.9\n",
      "chop_filtered_combined_edna_3.group----> 45 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 140 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 4 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 133 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_3.group----> similarity = 0.89\n",
      "chop_filtered_combined_edna_3.group----> 25 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 109 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 2 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 110 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 3 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> similarity = 0.88\n",
      "chop_filtered_combined_edna_3.group----> 19 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 85 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 23 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 63 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 1 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "---finetune group 0 cycle 1\n",
      "---finetune group 0 cycle 2\n",
      "---finetune group 0 cycle 3\n",
      "chop_filtered_combined_edna_3.group----> 352 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> similarity = 0.87\n",
      "chop_filtered_combined_edna_3.group----> 277 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 80 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 34 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 46 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 5 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> similarity = 0.86\n",
      "chop_filtered_combined_edna_3.group----> 12 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 36 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 5 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> 31 comparisons to calculate\n",
      "...processing: file_0.todo\n",
      "chop_filtered_combined_edna_3.group----> no sequences passed criteria\n",
      "chop_filtered_combined_edna_3.group----> similarity = 0.85\n",
      "chop_filtered_combined_edna_3.group----> 2 sequences added to the groups\n",
      "chop_filtered_combined_edna_3.group----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_14.group containing 10 sequences.\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_23.group containing 8 sequences.\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "reading chop_filtered_combined_edna_19.group containing 7 sequences.\n",
      "--> Number of groups after removing groups with less than 4 sequences: 0\n",
      "----> Making consensus for each group\n",
      "...making consensuses file_0.todo\n",
      "--> Nothing to do\n",
      "Writing sequences with high similarity in separate files\n",
      "-----------------------------------------------------------\n",
      "amplicon_sorter version: 2023-03-24\n",
      "-----------------------------------------------------------\n",
      "- date and time = March 27, 2023, 10:32PM\n",
      "- input file = /home/exouser/edna_analysis/filtered_reads/chop_filtered_combined_edna.fastq\n",
      "- output folder = /home/exouser/edna_analysis/sorted_amplicons/chop_filtered_combined_edna\n",
      "- minlength = 300\n",
      "- maxlength = None\n",
      "- maxreads = 10000\n",
      "- allreads = False\n",
      "- n_processes = 8\n",
      "- similar_genes = 80.0\n",
      "- similar_species_groups = 90 (Estimated)\n",
      "- similar_species = 85.0\n",
      "- similar_consensus = 96.0\n",
      "- length_difference_consensus = 8.0\n",
      "- save_fastq = False\n",
      "- random = True\n",
      "- compare_all = False\n",
      "- histogram_only = False\n",
      "- species_only = False\n",
      "-----------------------------------------------------------\n",
      "--> chop_filtered_combined_edna_1_0.fasta contains 43 sequences (27.04%)\n",
      "43/159 sequences assigned in groups for chop_filtered_combined_edna_1.group (27.04%)\n",
      "0/17 sequences assigned in groups for chop_filtered_combined_edna_13.group (0.0%)\n",
      "0/19 sequences assigned in groups for chop_filtered_combined_edna_15.group (0.0%)\n",
      "0/11 sequences assigned in groups for chop_filtered_combined_edna_11.group (0.0%)\n",
      "0/53 sequences assigned in groups for chop_filtered_combined_edna_6.group (0.0%)\n",
      "0/62 sequences assigned in groups for chop_filtered_combined_edna_2.group (0.0%)\n",
      "0/6 sequences assigned in groups for chop_filtered_combined_edna_10.group (0.0%)\n",
      "0/28 sequences assigned in groups for chop_filtered_combined_edna_12.group (0.0%)\n",
      "0/7 sequences assigned in groups for chop_filtered_combined_edna_7.group (0.0%)\n",
      "0/11 sequences assigned in groups for chop_filtered_combined_edna_21.group (0.0%)\n",
      "0/7 sequences assigned in groups for chop_filtered_combined_edna_17.group (0.0%)\n",
      "0/6 sequences assigned in groups for chop_filtered_combined_edna_25.group (0.0%)\n",
      "0/6 sequences assigned in groups for chop_filtered_combined_edna_16.group (0.0%)\n",
      "0/6 sequences assigned in groups for chop_filtered_combined_edna_22.group (0.0%)\n",
      "0/25 sequences assigned in groups for chop_filtered_combined_edna_9.group (0.0%)\n",
      "0/8 sequences assigned in groups for chop_filtered_combined_edna_26.group (0.0%)\n",
      "--> chop_filtered_combined_edna_5_0.fasta contains 74 sequences (96.1%)\n",
      "74/77 sequences assigned in groups for chop_filtered_combined_edna_5.group (96.1%)\n",
      "--> chop_filtered_combined_edna_0_0.fasta contains 1131 sequences (29.94%)\n",
      "--> chop_filtered_combined_edna_0_1.fasta contains 427 sequences (11.3%)\n",
      "--> chop_filtered_combined_edna_0_2.fasta contains 104 sequences (2.75%)\n",
      "--> chop_filtered_combined_edna_0_3.fasta contains 128 sequences (3.39%)\n",
      "1790/3778 sequences assigned in groups for chop_filtered_combined_edna_0.group (47.38%)\n",
      "0/14 sequences assigned in groups for chop_filtered_combined_edna_20.group (0.0%)\n",
      "0/35 sequences assigned in groups for chop_filtered_combined_edna_8.group (0.0%)\n",
      "0/38 sequences assigned in groups for chop_filtered_combined_edna_4.group (0.0%)\n",
      "0/45 sequences assigned in groups for chop_filtered_combined_edna_18.group (0.0%)\n",
      "0/9 sequences assigned in groups for chop_filtered_combined_edna_24.group (0.0%)\n",
      "--> chop_filtered_combined_edna_3_0.fasta contains 349 sequences (76.2%)\n",
      "349/458 sequences assigned in groups for chop_filtered_combined_edna_3.group (76.2%)\n",
      "0/10 sequences assigned in groups for chop_filtered_combined_edna_14.group (0.0%)\n",
      "0/8 sequences assigned in groups for chop_filtered_combined_edna_23.group (0.0%)\n",
      "0/7 sequences assigned in groups for chop_filtered_combined_edna_19.group (0.0%)\n",
      "(amplicon_sorter) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "python3 amplicon_sorter.py\\\n",
    " -i /home/exouser/edna_analysis/filtered_reads/chop_filtered_combined_edna.fastq\\\n",
    " -o ~/edna_analysis/sorted_amplicons\\\n",
    " --nprocesses 8\\\n",
    " --random"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
